core/test/element_math_test.cpp

```diff
@@ -46,6 +47,28 @@ TEST(ElementNormTest, value) {
   EXPECT_EQ(element::norm(v2), 5);
 }
 
+TEST(ElementPowTest, unit) {
+  // element::pow cannot handle units itself as that requires the unit of the
+  // base but the *value* of the exponent. This does not fit into the usual
+  // transform framework.
+  EXPECT_EQ(element::pow(units::one, units::one), units::one);
+  EXPECT_THROW_DISCARD(element::pow(units::one, units::m), except::UnitError);
+  EXPECT_THROW_DISCARD(element::pow(units::s, units::one), except::UnitError);
```

Comment:
Why is this not OK? Exponent is dimensionless.

Reply:
`element::pow` is unable to determine the correct unit because it needs the _value_ of the exponent and not just its unit. I chose to make it throw whenever any argument is not dimensionless to indicate that the caller has to take care of units. This could be changed to allow calls like line 56. Maybe this would even simplify `variable::pow`...